### PR TITLE
fix: MirNetv2 Implementation

### PR DESCRIPTION
**Pull request by soumik12345**
Changes proposed by this PR:

- [x] Fixes a few crucial bugs in MirNetv2
- [x] Adds serialization logic to MirNetv2
- [x] Fix crucial bugs in Evaluator

### CODE REVIEW

File: restorers/model/mirnetv2/downsample.py [hunk 1]
```diff
@@ -1,72 +1,76 @@
...
+
+    Args:
+        channels (int): number of input channels.
+        scale_factor (int): number of downsample operations.
+        channel_factor (float): factor by which number of the number of output channels vary.
     """
 
     def __init__(
-        self, channels: int, scale_factor: float, channel_factor: float, *args, **kwargs
+        self, channels: int, scale_factor: int, channel_factor: float, *args, **kwargs
```
Comment (Samarendra109) [inline restorers/model/mirnetv2/downsample.py:53]: scale_factor should be float

**Review by Samarendra109 — CHANGES_REQUESTED** [review]
I have requested for typing addition and adding get_config method. @soumik12345 please take a look.

File: restorers/model/mirnetv2/mirnet.py [hunk 2]
```diff
@@ -50,47 +66,64 @@ def get_config(self) -> Dict:
...
 
         self.conv_out = tf.keras.layers.Conv2D(3, kernel_size=3, padding="same")
 
-    def call(
-        self,
-        inputs: tf.Tensor,
-        training: Optional[bool] = None,
-        mask: Optional[bool] = None,
-    ) -> tf.Tensor:
+    def call(self, inputs: tf.Tensor, training=None, mask=None) -> tf.Tensor:
```
[the remark this answers is not in the document]
Comment (soumik12345) [reply restorers/model/mirnetv2/mirnet.py:126]: It's not an issue.

File: restorers/model/mirnetv2/upsample.py [hunk 3]
```diff
@@ -5,66 +5,53 @@
...
+    Args:
+        channels (int): number of input channels.
+        scale_factor (int): number of downsample operations.
+        channel_factor (float): factor by which number of the number of output channels vary.
     """
 
     def __init__(
-        self, channels: int, scale_factor: float, channel_factor: float, *args, **kwargs
-    ) -> None:
+        self, channels: int, scale_factor: int, channel_factor: float, *args, **kwargs
```
[the remark this answers is not in the document]
Comment (soumik12345) [reply restorers/model/mirnetv2/upsample.py:46]: Yep.

**Review by Samarendra109 — APPROVED** [review]
LGTM

**Pull request merged**